pancreabble/urchin.py

```diff
@@ -123,19 +124,19 @@ def main(self, args, app):
         cgm_history = json.loads(open(args.glucose_history).read())
         cgm_records = [
             {
-                'date': datetime.strptime(r['display_time'], '%Y-%m-%dT%H:%M:%S'),
-                'sgv': r['glucose'],
-                'trend': DEXCOM_TRENDS.index(r.get('trend_arrow')),
+                'date': dateutil.parser.parse(r['dateString'], ignoretz=True),
+                'sgv': r['sgv'],
+                'trend': DEXCOM_TRENDS.index(r.get('direction').upper().replace("FORTYFIVE","45_"))
```

Comment:
1. This is brittle and difficult to understand.

    You should replace `DEXCOM_TRENDS` with a dict:
    ```python
    DIRECTION_TO_TREND_NUMBER = {
        'DoubleUp': 1,
        'DOUBLE_UP': 1,
        'SingleUp': 2,
        'SINGLE_UP': 2,
        '45_UP': 3,
        'FortyFiveUp': 3,
        ...
    }
    ```

    and change this line to `DIRECTION_TO_TREND_NUMBER.get(r.get('direction'))`.

2. Can you add a comment with a link to the code which produces these new direction strings?

Reply:
This is suggested by @applehat for use with xdrip see https://github.com/openaps/oref0/pull/570#issuecomment-318553619

I'll ask him the source, because I can't find the direction code at this moment.